[SimWPEric] iP

SimWPEric opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
* easy to learn
+ ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/50).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :stuck_out_tongue_winking_eye:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

SimWPEric committed:
Added greet, exit

SimWPEric committed:
Added echo

SimWPEric committed:
Added add, list

SimWPEric committed:
Added mark as done, mark as not done

SimWPEric committed:
Added ToDos, Event, Deadline

SimWPEric committed:
Added Text UI Testing

SimWPEric committed:
Added error handling

SimWPEric committed:
Added delete

SimWPEric committed:
Added enums

SimWPEric committed:
Fix indentation

SimWPEric committed:
Add save feature

SimWPEric committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add save feature

SimWPEric committed:
Replace String-Based Date with LocalDate Objects

SimWPEric committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Replace String-Based Date with LocalDate Objects

SimWPEric committed:
Add Parser class to handle user command

SimWPEric committed:
Add constructor to Duke class

This commit introduces a constructor to the Duke class that takes a `filePath` parameter.

SimWPEric committed:
Add Ui class to handle interactions with user

SimWPEric committed:
Merge branch 'A-MoreOOP'

* A-MoreOOP:
  Add Ui class to handle interactions with user
  Add constructor to Duke class
  Add Parser class to handle user command

SimWPEric committed:
Divide classes into packages

SimWPEric committed:
Merge branch 'A-Packages'

* A-Packages:
  Divide classes into packages

SimWPEric committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552' into A-Gradle

* commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552':
  Bump gradle and lib version
  Add Gradle support

SimWPEric committed:
Add gradle support

SimWPEric committed:
Merge branch 'A-Gradle'

* A-Gradle:
  Add gradle support
  Bump gradle and lib version
  Add Gradle support

SimWPEric committed:
Add JUnit tests for Todo, Deadline and Event task

SimWPEric committed:
Merge branch 'A-JUnit'

* A-JUnit:
  Add JUnit tests for Todo, Deadline and Event task

SimWPEric committed:
Add find feature

SimWPEric committed:
Add JavaDoc comments

Add header comments to all non-private classes/methods, and non-trivial private methods.

SimWPEric committed:
Fix coding standard violations

SimWPEric committed:
Add JavaDoc for Ui class

SimWPEric committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find feature

SimWPEric committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc for Ui class
  Add JavaDoc comments

# Conflicts:
#	src/main/java/duke/Command.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java

SimWPEric committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Fix coding standard violations

SimWPEric committed:
Extract Command classes

SimWPEric committed:
Add checkstyle configs

SimWPEric committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Add checkstyle configs

SimWPEric committed:
Add simple GUI

SimWPEric committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add simple GUI

simbayippy commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,78 @@
+        );
+
+        // Closes the GUI
+        if (input.equals("bye")) {
```
what if user input is "BYE"?

SimWPEric replied:
Thank you for pointing this out, I have added `input.toLowerCase()` to make the exit command case insensitive.

simbayippy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,166 @@
+     * @throws DukeInvalidDateFormatException If the input contains an invalid date format.
+     * @throws DukeEndDateBeforeStartDateException If an event's end date is before its start date.
+     */
+    public static Command parseUserInput(String input) throws DukeInvalidArgumentException,
```
similarly what is users input are not all lowercase?

SimWPEric replied:
Yes, you are right, I have added additional format specifiers ie: `String todoPattern = "(?i)todo (.+)";`, where `(?i)` makes the command case insensitive when parsing user input.

simbayippy commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,156 @@
+    private void writeTaskListToFile(TaskList taskList) throws IOException {
+        Path filePath = Paths.get(this.filepath);
+        try {
+            Files.deleteIfExists(filePath);
```
is this unnecessary? when doing File file = new File(filePath); i believe it should already override the existing file in the filepath

SimWPEric replied:
My idea is to overwrite the existing file rather than appending to an existing file. Please let me know if you have any alternative implementations !

simbayippy reviewed (commented):
Overall your code is brilliantly done. Abstraction of classes are done very clearly. Took away some learning points from your code too - thank you!

SimWPEric committed:
Update parser to accept case insensitive user inputs

SimWPEric committed:
Fix JavaDoc style violations

Changes Made:
- Updated JavaDoc comments in classes and methods within the command package.
- Enhanced JavaDoc comments in classes and methods within the ui package.

SimWPEric committed:
Add assert statements to validate assumptions

1. In `Event.java`:
Validate the assumption that the start date is before the end date when
creating an Event object. This assertion will catch potential logic errors
during development, ensuring that events are correctly defined with
valid date ranges.

2. In `TaskList.java`:
Verify the invariant that a task is successfully removed from the taskList
when using the `delete` method. This assertion serves to maintain the
integrity of the task list, ensuring that tasks are correctly removed and
preventing unexpected issues.

SimWPEric committed:
Refactor code to replace for loops with Java streams

Using stream in place of loops should make our code simpler,
more elegant, and less bug-prone.

SimWPEric committed:
Refactor Parser class

The Parser class contains long methods with 96 LOC.

Refactoring is necessary to improve readability.

The code is refactored to separate parsing methods for each command
type.

Separating parsing logic into dedicated methods improves code
structure and allows for easier maintenance and future extensions.

This refactoring does not introduce any function changes but
enhances code quality and maintainability.

SimWPEric committed:
Merge pull request #1 from SimWPEric/branch-A-Assertions

Add assert statements to validate assumptions

SimWPEric committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assert statements to validate assumptions

SimWPEric committed:
Merge branch 'master' into branch-A-Assertions

* master:

SimWPEric committed:
Merge pull request #2 from SimWPEric/branch-A-Streams

Refactor code to replace for loops with Java streams

SimWPEric committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Refactor code to replace for loops with Java streams

SimWPEric committed:
Merge pull request #3 from SimWPEric/branch-A-CodeQuality

Refactor Parser class

SimWPEric committed:
Update DialogBox

Replaced DialogBox with DukeDialog and UserDialog for individual styling.

SimWPEric committed:
Create gradle.yml

SimWPEric committed:
Fix styling errors

SimWPEric committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:

SimWPEric committed:
Add TaskList sort functionality

TaskList now contains Tasks sorted in the
order of Todo, Deadline, Event.

SimWPEric committed:
Merge branch 'branch-C-Sort'

* branch-C-Sort:
  Add TaskList sort functionality

SimWPEric committed:
Add Ui Screenshot

SimWPEric committed:
Update user guide

SimWPEric committed:
Update user guide to include ToC

SimWPEric committed:
Fix platform exit bug

SimWPEric committed:
Update build.gradle

SimWPEric committed:
Revert build.gradle
